### DISCUSSION TITLE
DWMWindow frame highlights

### DISCUSSION THREAD

**Discussion by kfh83** [Maplespe/DWMBlurGlass#295] [Ideas]
![image](https://github.com/Maplespe/DWMBlurGlass/assets/85626296/6799263d-039d-4c9b-9cae-06e7b2bb5b3c)
_(highlights in Windows 7, highlighted in green. The right side top highlights seem to be mirrored from the left side by DWM itself as there are no dedicated images for them.)_

They can currently be "achieved" in Windows 10 by putting them together with the border image, however this results in them being stretched out if the titlebar is too tall, aswell as full glass apps having the top half of them be brighter than the bottom half. 
![image](https://github.com/Maplespe/DWMBlurGlass/assets/85626296/7a015211-3e6d-4fdf-8fa9-227b74164436)
_(full glass app, Windows 10 with "baked" highlights)_
![image](https://github.com/Maplespe/DWMBlurGlass/assets/85626296/1943fe6d-0948-4f69-abf2-b3c3fe182f11)
_(gadgets app, Windows 10 with "baked" highlights)_

![image](https://github.com/Maplespe/DWMBlurGlass/assets/85626296/fd6c3e19-e249-46cf-9f28-e964ae838c52)
_(full glass app, Windows 7)_
![image](https://github.com/Maplespe/DWMBlurGlass/assets/85626296/12850520-0a2b-4fd4-94b9-1aaf97287e3f)
_(gadgets app, Windows 7)_

The side highlights only seem to render when there's a content area, while the top ones seem to always be rendered (the exception to this seems to be the taskbar/start menu, not sure of other exceptions but these are the notable ones). They have their own sizing margins independent from the actual window frame, and it seems like the top highlights are a fixed size, seem to go inside eachother when the window is small enough and bleed into the bottom frames if the window is short enough. I should also probably mention they have different opacities for active/inactive windows.

On another note, if this does end up getting added on DWMBlurGlass, could it atleast be put outside the "Windows 7-style captions" option? I understand wanting to be an all-in-one app but i think it would be nice to not break compatibility with other customization software, especially when it seems like some of these features (namely caption button glow) could very well be put outside of that option and still work fine with or without it.

**Comment by OldManAlpha** [comment]
Thanks for looking into specifically how that works. I really hope a solution can be made for this issue.

**Comment by aubymori** [comment]
I believe the corner highlights not showing on the taskbar and start menu has to do with the DWM frame effect there being applied with the `DwmEnableBlurBehindWindow` API. Not sure exactly what these conditions are, but it's a start.

**Reply by Skyyblaze** [reply]
I did notice some oddities with DwmEnableBlurBehindWindow too. It seems DWMBlurGlass has some effect on it but I'm not sure how exactly it interacts. I shared my observations here: https://github.com/Maplespe/DWMBlurGlass/discussions/287

**Comment by TorutheRedFox** [comment]
quick note: vista did have a dedicated image for it but it too was just a mirrored version of the left side, so they just got rid of it in 7 as an optimization
